packages/aws-cdk-lib/aws-dynamodb/lib/global-table-base.ts

```diff
@@ -0,0 +1,488 @@
+    metricNameMapper?: (op: Operation) => string) {
+    const metrics: Record<string, IMetric> = {};
+    const mapper = metricNameMapper ?? (op => op.toLowerCase());
+
+    if (props?.dimensions?.Operation) {
+      throw new Error('Invalid properties. Operation dimension is not supported when calculating operational metrics');
+    }
+
+    for (const operation of operations) {
+      const metric = this.metric(metricName, {
+        ...props,
+        dimensionsMap: { TableName: this.tableName, Operation: operation, ...props?.dimensions },
+      });
+
+      const operationMetricName = mapper(operation);
+      const firstChar = operationMetricName.charAt(0);
+      if (firstChar === firstChar.toUpperCase()) {
+        throw new Error(`Mapper generated an illegal operation metric name: ${operationMetricName}. Must start with a lowercase letter`);
+      }
+
+      metrics[operationMetricName] = metric;
+    }
+
+    return metrics;
+  }
+
+  /**
+   * Adds an IAM policy statement associated with this Global Table to an IAM principal's policy.
+   *
+   * @param grantee the principal (no-op if undefined)
+   * @param options options for keyActions, tableActions, and streamActions
+   */
+  private combinedGrant(grantee: IGrantable, options: { keyActions?: string[], tableActions?: string[], streamActions?: string[] }) {
+    if (options.keyActions) {
+      for (const key of Object.values(this.replicaKeys)) {
+        key.grant(grantee, ...options.keyActions);
+      }
+      this.encryptionKey && this.encryptionKey.grant(grantee, ...options.keyActions);
+    }
+
+    if (options.tableActions) {
+      const resources = [
+        this.tableArn,
+        this.hasIndex ? `${this.tableArn}/index/*` : Aws.NO_VALUE,
+        ...this.replicaArns,
+        ...this.replicaArns.map(arn => this.hasIndex ? `${arn}/index/*` : Aws.NO_VALUE),
+      ];
+      return Grant.addToPrincipal({
+        grantee,
+        actions: options.tableActions,
+        resourceArns: resources,
+        scope: this,
+      });
+    }
+
+    if (options.streamActions) {
+      if (this.streamArns.length === 0) {
+        throw new Error(`No stream ARNs found on the table ${this.node.path}`);
+      }
+
+      return Grant.addToPrincipal({
+        grantee,
+        actions: options.streamActions,
+        resourceArns: this.streamArns,
+        scope: this,
+      });
+    }
+
+    throw new Error(`Unexpected 'action', ${options.tableActions || options.streamActions}`);
+  }
+
+  private configureMetric(props: MetricProps) {
+    if (!props?.region && !props?.account) {
+      return new Metric({
+        region: this.region,
+        account: this.stack.account,
+        ...props,
+      });
+    }
+
+    if (!props?.region) {
+      return new Metric({
+        region: this.region,
+        ...props,
+      });
+    }
+
+    return new Metric({
+      account: this.stack.account,
+      ...props,
+    });
+  }
```

Comment:
Looks like the defaults are the same in each case, so this can be simplified?

```suggestion
  private configureMetric(props: MetricProps) {
    return new Metric({
      ...props,
      region: props?.region ?? this.region,
      account: props?.account ?? this.stack.account,
    });
  }
```

(I'm doing `...props` before instead of after, because if someone passes `({ region: undefined })` then the `undefined` will overwrite the value we just defaulted again. [Try it out](https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABAZwA4BsCGUBCBTYOAJzwAo5UpkAuRAJTwmIBMAeZKImMAcwBpE4ANZg4AdzAA+AJSIA3gChEiJmGRx0eAHTo4PUouXKtJilT5KjhOLQDk12xeUBfaQG4FzhQtCRYCFAxsAEFgKDwickoaekYWdk5ufkEwEXEpWUMVBHVNHT0DS2VrOwcnIxMtM2Ry1w8vBTQsXAJiMjlEACNMIjtuoltEOsagqFDwyI7+vp7B4ZHm-EISA0QSlOYCbjxmAWnEW3659wWQsIjV9fBN4G3drp6ZgaH3RAB6N8RWAFpvxDgKMggA)

Reply:
Way simpler! Updated.